pkg/agent/endpoints/workload/handler.go

```diff
@@ -178,6 +180,22 @@ func (h *Handler) ValidateJWTSVID(ctx context.Context, req *workload.ValidateJWT
 	}
 	log.WithField(telemetry.SPIFFEID, spiffeID).Debug("Successfully validated JWT")
 
+	id, err := spiffeid.FromString(spiffeID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unexpected SPIFFE ID: %v", err)
+	}
+
+	if !id.MemberOf(h.c.TrustDomain) {
```

Comment:
Does this imply these claims are trusted from any foreign JWT trust domain? Is that desirable, or do we want to scope claims by trust domains?

Reply:
it imply that only `"sub", "exp", "aud"` are allowed for foreign JWTs except a configuration is expanding the `allowed` list.
if JWT is from a Agent's TD, we return every claim